### PR TITLE
1) Added tests for all classes

### PR DESCRIPTION
**Pull request by pasha149**
2) Corrected gaffes in several classes

### CODE REVIEW

File: src/classes/CustomerHandlerTest.cls [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+			Request_token_verifier__c = '', 
+			Access_Token_Secret__c = '', 
+			Request_Token_Secret__c = ''
+		);
+	}
+
+	@isTest private static void test_getRecords() {
+		init();
+		Contact testContact = new Contact();
+		RestClient getInstance = RestClient.getInstance;
```
Comment (maximcodeswat) [inline src/classes/CustomerHandlerTest.cls]: This is required for the unit test?

File: src/classes/CustomerHandlerTest.cls [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			AccountId = testAccount.Id
+		);
+		insert testContact;
+
+		testContact = [
+			SELECT Name, Email, Account.Name, Account.Magento_Address_Id__c, Customer_Id__c, FirstName, LastName
+			FROM Contact
+		];
+
+		RestClient getInstance = RestClient.getInstance;
```
Comment (maximcodeswat) [inline src/classes/CustomerHandlerTest.cls]: This is required for the unit test?

File: src/classes/CustomerSyncHandlerTest.cls [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+			AccountId = testAccount.Id
+		);
+		insert testContact;
+
+		testContact = [
+			SELECT Name, Email, Account.Name, Account.Magento_Address_Id__c, Customer_Id__c, FirstName, LastName
+			FROM Contact
+		];
+		RestClient getInstance = RestClient.getInstance;
+		getInstance.setAuthorizator(Oauth1AuthorizationClient.oauthClient);
```
Comment (maximcodeswat) [inline src/classes/CustomerSyncHandlerTest.cls]: This is required for the unit test?

File: src/classes/CustomerSyncHandlerTest.cls [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+		CustomerSyncHandler customerSyncHandler = new CustomerSyncHandler(1);
+		
+		Test.stopTest();
+
+		System.assertEquals('11', customerSyncHandler.execute());
+	}
+	@isTest private static void test_executeCustomerId() {
+		init();
+		RestClient getInstance = RestClient.getInstance;
```
Comment (maximcodeswat) [inline src/classes/CustomerSyncHandlerTest.cls]: This is required for the unit test?

File: src/classes/CustomerSyncHandlerTest.cls [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+			Customer_Id__c = '11',
+			AccountId = testAccount.Id
+		);
+		insert testContact;
+
+		testContact = [
+			SELECT Name, Email, Account.Name, Account.Magento_Address_Id__c, Customer_Id__c, FirstName, LastName
+			FROM Contact
+		];
+		RestClient getInstance = RestClient.getInstance;
```
Comment (maximcodeswat) [inline src/classes/CustomerSyncHandlerTest.cls]: This is required for the unit test?

File: src/classes/GlobalBatchTest.cls [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+		Test.stopTest();
+
+		AsyncApexJob examplBatchIterable = [
+			SELECT Status
+			FROM AsyncApexJob
+			WHERE id = :idApexJobBatchIterable
+			LIMIT 1
+		];
+
+		System.assertEquals('Completed', examplBatchIterable.Status);
```
Comment (maximcodeswat) [inline src/classes/GlobalBatchTest.cls]: I would check created records on Salesforce side.

File: src/classes/OrderSyncHandlerTest.cls [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+			CloseDate = Date.today() + 10,
+			Order_Id__c = 'Test Order'
+		);
+		insert testOpportunity;
+
+		Test.setMock(HttpCalloutMock.class, new MockHttpResponseGenerator());
+
+		Test.startTest();
+
+		OrderSyncHandler ordererSyncHandler = new OrderSyncHandler();
```
Comment (maximcodeswat) [inline src/classes/OrderSyncHandlerTest.cls]: ordererSyncHandler. Could you check the name of the variable?

File: src/classes/CustomerHandlerTest.cls [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+		init();
+		Contact testContact = new Contact(
+			FirstName = 'Name',
+			LastName = 'LastName',
+			Email = 'test@test.com',
+			Customer_Id__c = 'Test Id'
+		);
+		insert testContact;
+
+		RestClient getInstance = RestClient.getInstance;
```
Comment (maximcodeswat) [inline src/classes/CustomerHandlerTest.cls]: Is it required for this method?

File: src/classes/GlobalBatchTest.cls [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+
+		Test.stopTest();
+
+		List<Contact> newContacts = [
+			SELECT Id
+			FROM Contact
+			LIMIT 10000
+		]; 
+
+		System.assertEquals(true, newContacts.size() > 0);
```
Comment (maximcodeswat) [inline src/classes/GlobalBatchTest.cls]: I would suggest to use the following code: System.assert( ! newContacts.isEmpty());

File: src/classes/OrderSyncHandlerTest.cls [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+		ordererSyncHandler.execute();
+		
+		Test.stopTest();
+
+		Opportunity order = [
+			SELECT Name
+			FROM Opportunity
+			LIMIT 1
+		];
+		System.assertEquals(true, order.Name == 'Test Opportunity');
```
Comment (maximcodeswat) [inline src/classes/OrderSyncHandlerTest.cls]: System.assert(order.Name == 'Test Opportunity');

File: src/classes/ProductHandlerTest.cls [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+		Contact testContact = new Contact(
+			FirstName = 'Name',
+			LastName = 'LastName',
+			Email = 'test@test.com',
+			Customer_Id__c = 'Test Id',
+			AccountId = testAccount.Id
+		);
+		insert testContact;
+
+		testContact = [
```
Comment (maximcodeswat) [inline src/classes/ProductHandlerTest.cls]: Why should we select this testContact record?

File: src/classes/ProductSyncHandlerTest.cls [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+		
+		Test.stopTest();
+
+		System.assertEquals('1', productSyncHandler.execute());
+	}
+
+	@isTest private static void test_executeExistingProduct() {
+		init();
+
+		RestClient getInstance = RestClient.getInstance;
```
Comment (maximcodeswat) [inline src/classes/ProductSyncHandlerTest.cls]: Is it required for this method?

File: src/classes/CustomerHandlerTest.cls (hunk 8 above)
Comment (pasha149) [inline src/classes/CustomerHandlerTest.cls]: It's required for test coverage (class CustomerHandler, line 58).

**Pull request merged**